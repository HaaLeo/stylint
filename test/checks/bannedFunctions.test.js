'use strict';

const bannedFunctions = require('../../src/checks/bannedFunctions');

describe('bannedFunctions: ban use of specific key words', () => {
  let context;

  beforeEach(() => {
    context = {
      report: jest.fn(),
    };
  });

  it("don't report if a line doesn't have any banned functions", () => {
    bannedFunctions(Object.assign(context, { line: '.foo' }));
    expect(context.report).not.toHaveBeenCalled();
  });

  it("don't report if a line has banned functions but is not found", () => {
    bannedFunctions(Object.assign(context, { line: '.foo', config: ['translate3d'] }));
    expect(context.report).not.toHaveBeenCalled();
  });

  it('report if line has a banned function', () => {
    bannedFunctions(Object.assign(context, { line: 'translate3d(1px, 1px, 0px)', config: ['translate3d'] }));
    expect(context.report.mock.calls).toMatchSnapshot();
  });
});
