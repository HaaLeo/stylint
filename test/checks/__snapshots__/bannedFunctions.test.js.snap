// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bannedFunctions: ban use of specific key words report if line has a banned function 1`] = `
Array [
  Array [
    Object {
      "column": 0,
      "message": "unexpected banned function translate3d encountered",
    },
  ],
]
`;
